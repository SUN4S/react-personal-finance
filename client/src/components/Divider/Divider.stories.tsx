import { Meta, Story } from "@storybook/react/types-6-0";

import { Divider } from "./Divider";

interface TemplateProps {
  theme: string;
  text: string;
}

export default {
  title: "Components/Divider",
  component: Divider,
} as Meta;

const Template: Story<TemplateProps> = (args) => {
  return (
    <html data-theme={args.theme}>
      <div
        style={{
          backgroundColor: "var(--sidebar-bg)",
          padding: "8px",
        }}
      >
        <Divider text={args.text} />
      </div>
    </html>
  );
};

export const Default_Light = Template.bind({});
Default_Light.args = {
  theme: "light",
  text: "OR",
};

export const Dark_Light = Template.bind({});
Dark_Light.args = {
  theme: "dark",
  text: "OR",
};
