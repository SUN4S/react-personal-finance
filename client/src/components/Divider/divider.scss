.divider {
  width: 100%;
  display: flex;
  align-items: center;
  text-align: center;
  color: var(--text-secondary);
  margin: 12px 0;

  &:before,
  &:after {
    content: "";
    flex: 1;
    border-bottom: 1px solid var(--text-secondary);
  }

  &:not(:empty)::before {
    margin-right: 6px;
  }

  &:not(:empty)::after {
    margin-left: 6px;
  }
}
