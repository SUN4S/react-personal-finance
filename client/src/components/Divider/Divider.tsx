import "./Divider.scss";

interface Props {
  text: string;
}

// Simple divider, so far only used in login
export const Divider = ({ text }: Props) => {
  return (
    <div className="divider" data-testid="divider">
      {text}
    </div>
  );
};
