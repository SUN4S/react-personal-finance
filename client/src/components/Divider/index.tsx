import "./divider.scss";

// Simple divider, so far only used in login
export const Divider = () => {
  return <div className="divider">OR</div>;
};
