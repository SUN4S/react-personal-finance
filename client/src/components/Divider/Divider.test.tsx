/**
 * @jest-environment jsdom
 */

import { cleanup, render } from "@testing-library/react";

import { Divider } from "./Divider";
import renderer from "react-test-renderer";

describe("Divider Component", () => {
  afterAll(cleanup);

  it("Renders Divider", () => {
    const { getByTestId } = render(<Divider text="OR" />);
    const dividerElement = getByTestId("divider");
    expect(dividerElement).toBeTruthy();
  });

  it("Renders Divider Text", () => {
    const { getByTestId } = render(<Divider text="OR" />);
    const dividerElement = getByTestId("divider");
    expect(dividerElement).toHaveTextContent("OR");
    expect(dividerElement).toHaveCss
  });

  it("Matches Snapshot", () => {
    const tree = renderer.create(<Divider text="OR" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
