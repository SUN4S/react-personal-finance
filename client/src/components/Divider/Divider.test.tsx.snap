// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider Component Matches Snapshot 1`] = `
<div
  className="divider"
  data-testid="divider"
>
  OR
</div>
`;
